septima-js-generator/src/main/java/com/septima/generator/EntitiesSnapshots.java

```diff
@@ -0,0 +1,161 @@
+                case DATE:
+                    return "Date";
+                case BOOLEAN:
+                    return "Boolean";
+                default:
+                    return "String";
+            }
+        } else {
+            return "String";
+        }
+    }
+
+    /**
+     * Returns a {@link Map.Entry} with specified key and value.
+     * Unlike {@link Map#entry(Object, Object)} retains {@code null} values to preserve datum structure.
+     *
+     * @param aKey   A key.
+     * @param aValue A value.
+     * @param <K>    Map's key type.
+     * @param <V>    Map's value type.
+     * @return A {@link Map.Entry} with specified key and value.
+     */
+    public static <K, V> Map.Entry<K, V> entry(K aKey, V aValue) {
+        return new AbstractMap.SimpleEntry<>(aKey, aValue);
+    }
+
+    /**
+     * Returns map with specified entries.
+     * Unlike {@link Map#ofEntries(Map.Entry[])} retains {@code null} values to preserve datum structure.
+     *
+     * @param aEntries An array of entries.
+     * @param <K>      Map's key type.
+     * @param <V>      Map's value type.
+     * @return A map with specified entries.
+     * Unlike {@link Map#ofEntries(Map.Entry[])} retains {@code null} values to preserve datum structure.
+     */
+    public static <K, V> Map<K, V> map(Map.Entry<K, V>... aEntries) {
+        Map<K, V> map = new HashMap<>();
+        for (Map.Entry<K, V> e : aEntries) {
+            Objects.requireNonNull(e.getKey());
+            map.put(e.getKey(), e.getValue());
+        }
+        return Collections.unmodifiableMap(map);
+    }
+
+    private void toSnapshotJson(Path sqlEntityFile) throws IOException {
+        Path entityRelativePath = entities.getEntitiesRoot().relativize(sqlEntityFile);
+        String entityRelativePathName = entityRelativePath.toString().replace('\\', '/');
+        String entityRef = entityRelativePathName.substring(0, entityRelativePathName.length() - 4);
+        SqlEntity entity = entities.loadEntity(entityRef);
+
+        Path jsonFile = considerSnapshotJson(sqlEntityFile);
+        if (!jsonFile.getParent().toFile().exists()) {
+            jsonFile.getParent().toFile().mkdirs();
+        }
+        JSON_WRITER.writeValue(jsonFile.toFile(), map(
+                entry("title", entity.getTitle()),
+                entry("source", entities.dataSourceOf(entity.getDatabase())),
+                entry("public", entity.isPublicAccess()),
+                entry("sql", entity.getCustomSqlText() != null && !entity.getCustomSqlText().isBlank() ? entity.getCustomSqlText() : entity.getSqlText()),
+                entry("procedure", entity.isProcedure()),
+                entry("command", entity.isCommand()),
+                entry("readonly", entity.isReadonly()),
+                entry("parameters", entity.getParameters().entrySet().stream().collect(Collectors.toMap(Function.identity(), e -> map(
+                        entry("type", textOf(e.getValue().getType())),

```

Comment:
In my opinion It should be converted to string for example by overrode `toString` method of enumeration. In current case you have not optimal performance.

Reply:
Good point! I should think about it a bit.